autoload/venu/native.vim

```diff
@@ -0,0 +1,107 @@
+function! s:parseMenuLine(line)
+  let l:match = matchlist(a:line, "\\v^( *)([^ ]+)\\s+(.{-})\\s*\\r?$")
+  if len(l:match) == 0
+    return []
+  endif
+  return [len(l:match[1]), l:match[2], l:match[3]]
+endfunction
+
+" Extract <TAB>=^I separated description from name if there exists one
+function! s:extractDesc(name)
+  let l:tmp = matchlist(a:name, '\v^(.{-})\^I(.{-})$')
+  if len(l:tmp) == 0
+    let l:name = a:name
+    let l:desc = ''
+  else
+    let l:name = l:tmp[1]
+    let l:desc = l:tmp[2]
+  endif
+  return [l:name, l:desc]
+endfunction
+
+" Extract &Shortcut and replace escaped ampersands (&&).
+function! s:extractShortcut(name)
+  " These are not some curse words, just some regex to match first unescaped &.
+  let l:tmp = matchend(a:name, "\\v(\\&)@<!%(\\&\\&)*(\\&)(\\&)@!")
+  if l:tmp < 0
+    let l:shortcut = ''
+    let l:name = a:name
+  else
+    let l:shortcut = a:name[l:tmp]
+    let l:name = substitute(
+    \   strpart(a:name, 0, l:tmp - 1).strpart(a:name, l:tmp),
+    \   "&&", "&", "g"
+    \ )
+  endif
+  return [l:name, l:shortcut]
+endfunction
+
+function! venu#native#list()
+  let l:menu_str = execute("menu")
+  let l:root = []
+  let l:prevdepth = 0
+
+  for l:line in split(l:menu_str, "\\n")
+    let l:line = s:parseMenuLine(l:line)
+    if len(l:line) > 0
+      if match(l:line[1], "\\v^\\d+$") == 0
+        let l:indent   = l:line[0]
+        let l:depth    = l:indent / 2
+        let l:priority = str2nr(l:line[1])
+
+        let l:tmp  = s:extractDesc(l:line[2])
+        let l:name = l:tmp[0]
+        let l:desc = l:tmp[1]
+
+        let l:tmp      = s:extractShortcut(l:line[2])
+        let l:name     = l:tmp[0]
+        let l:shortcut = l:tmp[1]
+
+        let l:menu = {
```

Comment:
You could already create a venu conform menu with `venu#create(l:name)`.
Although the question is whether in e.g. this example:
```
    500 Html
      500 Convert^I<C-E>2hf
          n    <Plug>(Riv2HtmlFile)
          v    <C-C><Plug>(Riv2HtmlFile)<C-\><C-G>
          s    <C-C><Plug>(Riv2HtmlFile)<C-\><C-G>
          o    <C-C><Plug>(Riv2HtmlFile)<C-\><C-G>
```
the `Convert` is actually a menu or an item.. If it is a menu then the cmds for the different modes `n`,`v`,`s` and `o` would be the items.
I'm not sure what your plan is.

Anyway, it would make sense to make use of the `venu#create` and `venu#addItem` "interface".

Let me know in case you need further functionality (apart from the abilty to unregister certain menus/submenus).

Reply:
Convert is an item for Venu I guess. I plan to default to normal mode command since by the time `:VenuPrint` is called vim would be in normal mode.

I'm adding a `venu#native#import` which will make necessary calls to `venu#native#list` (actually renamed to `venu#native#parseMenu`), `venu#create` and `venu#addItem`.

> Let me know in case you need further functionality (apart from the abilty to unregister certain menus/submenus).

Thanks, I will. So far I don't need any extra functionality apart from unregistering.